lib/matplotlib/artist.py

```diff
@@ -1270,7 +1270,7 @@ def _get_setters_and_targets(self):
             func = getattr(self.o, name)
             if not callable(func):
                 continue
-            nargs = len(inspect.getfullargspec(func).args)
+            nargs = len(inspect.signature(func).parameters)
```

Comment:
Technically, this is not the same because `Signature.parameters` contains everything.

I think
~~~
len([p for p in inspect.signature(func).parameters.values()
    if p.kind == inspect.Parameter.POSITIONAL_OR_KEYWORD])
~~~
is equivalent, but haven't checked for cases with variable positional args or kwonly.


Some example code which may be useful to play around with:
~~~
>>> from pprint import pprint
>>> import inspect
>>> def f(a, b, c=0, **kwargs):
...     pass
... 
>>> inspect.getfullargspec(f)
FullArgSpec(args=['a', 'b', 'c'], varargs=None, varkw='kwargs', defaults=(0,), kwonlyargs=[], kwonlydefaults=None, annotations={})
>>> pprint(inspect.signature(f).parameters)
mappingproxy(OrderedDict([('a', <Parameter "a">),
                          ('b', <Parameter "b">),
                          ('c', <Parameter "c=0">),
                          ('kwargs', <Parameter "**kwargs">)]))
~~~

Reply:
However, the only thing we use `nargs` for is in the next line to detect if there if it less than 2 to detect if this is something that is named like a setter, but does not take enough arguments to actually be a setter.

Reply:
For the following

`len(inspect.getfullargspec(func).args) < 2` is true but `len(inspect.signature(func).parameters) < 2` is false.

~~~
Line2D.set_data(self, *args)
Axes.set_prop_cycle(self, *args, **kwargs)
Figure.set_constrained_layout_pads(self, **kwargs)
~~~

I did not check the full code path, but I assume they will be considered setters after the change - not sure if that's actually desired or not, but it's a change.

Reply:
fair enough, will make the changes!

Thanks for pushing on this.